diff --git a/Silicon/NVIDIA/Library/DxeDtPlatformDtbLoaderLib/DxeDtPlatformDtbKernelLoaderLib.c b/Silicon/NVIDIA/Library/DxeDtPlatformDtbLoaderLib/DxeDtPlatformDtbKernelLoaderLib.c
index ba009757..71593409 100644
--- a/Silicon/NVIDIA/Library/DxeDtPlatformDtbLoaderLib/DxeDtPlatformDtbKernelLoaderLib.c
+++ b/Silicon/NVIDIA/Library/DxeDtPlatformDtbLoaderLib/DxeDtPlatformDtbKernelLoaderLib.c
@@ -561,8 +561,7 @@ InstallFdt (
   }
 
   PlatformType = TegraGetPlatform ();
-  if ((PlatformType != TEGRA_PLATFORM_SILICON) ||
-      (PlatformResourceInfo->BootType == TegrablBootRcm))
+  if (TRUE) // Always use DTB from UEFI
   {
     Dtb = (VOID *)GetDTBBaseAddress ();
     if (fdt_check_header (Dtb) != 0) {
