diff --git a/cmd/nvidia-ctk/hook/update-ldcache/update-ldcache.go b/cmd/nvidia-ctk/hook/update-ldcache/update-ldcache.go
index 10fd839..4af3a70 100644
--- a/cmd/nvidia-ctk/hook/update-ldcache/update-ldcache.go
+++ b/cmd/nvidia-ctk/hook/update-ldcache/update-ldcache.go
@@ -87,8 +87,8 @@ func (m command) run(c *cli.Context, cfg *options) error {
 		return fmt.Errorf("failed to determined container root: %v", err)
 	}
 
-	ldconfigPath := m.resolveLDConfigPath("/sbin/ldconfig")
-	args := []string{filepath.Base(ldconfigPath)}
+	ldconfigPath := m.resolveLDConfigPath("/run/current-system/sw/bin/ldconfig")
+	args := []string{filepath.Base(ldconfigPath), "-C", "/etc/ld.so.cache", "-f",  "/etc/ld.so.conf"}
 	if containerRoot != "" {
 		args = append(args, "-r", containerRoot)
 	}
