diff --git a/Platform/NVIDIA/Jetson/Jetson.dsc b/Platform/NVIDIA/Jetson/Jetson.dsc
index 39e2c1d..3d6911d 100644
--- a/Platform/NVIDIA/Jetson/Jetson.dsc
+++ b/Platform/NVIDIA/Jetson/Jetson.dsc
@@ -33,3 +33,7 @@
 
 [PcdsFixedAtBuild]
   gNVIDIATokenSpaceGuid.PcdPlatformFamilyName|L"Jetson"
+!if $(CUSTOM_CAPSULE_CERT)
+  !include PublicCapsuleKey.cer.gEfiSecurityPkgTokenSpaceGuid.PcdPkcs7CertBuffer.inc
+  !include PublicCapsuleKey.cer.gFmpDevicePkgTokenSpaceGuid.PcdFmpDevicePkcs7CertBufferXdr.inc
+!endif
