diff -Naur source/0_Simple/matrixMul_nvrtc/Makefile source-new/0_Simple/matrixMul_nvrtc/Makefile
--- source/0_Simple/matrixMul_nvrtc/Makefile	2022-05-04 18:33:45.000000000 +0000
+++ source-new/0_Simple/matrixMul_nvrtc/Makefile	2023-02-01 22:43:15.652000000 +0000
@@ -390,8 +390,8 @@
 all: build
 
 build: matrixMul_nvrtc
-	$(EXEC) cp "$(CUDA_PATH)/$(CUDA_INSTALL_TARGET_DIR)include/cooperative_groups.h" .
-	$(EXEC) cp -r "$(CUDA_PATH)/$(CUDA_INSTALL_TARGET_DIR)include/cooperative_groups" .
+	$(EXEC) cp "$(CUDA_PATH)/include/cooperative_groups.h" .
+	$(EXEC) cp -r "$(CUDA_PATH)/include/cooperative_groups" .
 
 check.deps:
 ifeq ($(SAMPLE_ENABLED),0)
