diff -Naur a/lib/cmake/vpi/vpi-config.cmake b/lib/aarch64-linux-gnu/cmake/vpi/vpi-config.cmake
--- a/lib/aarch64-linux-gnu/cmake/vpi/vpi-config.cmake	1969-12-31 16:00:01.000000000 -0800
+++ b/lib/aarch64-linux-gnu/cmake/vpi/vpi-config.cmake	2022-09-27 14:30:22.957595096 -0700
@@ -42,16 +42,7 @@
 
 
 # Compute the installation prefix relative to this file.
-get_filename_component(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
-get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" REALPATH)
-
-get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
-get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
-get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
-get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
-if(_IMPORT_PREFIX STREQUAL "/")
-  set(_IMPORT_PREFIX "")
-endif()
+set(_IMPORT_PREFIX "@out@")
 
 # Create imported target vpi
 add_library(vpi SHARED IMPORTED)
