diff --git a/BaseTools/Plugin/LinuxGccToolChain/LinuxGccToolChain.py b/BaseTools/Plugin/LinuxGccToolChain/LinuxGccToolChain.py
index e7ef3df3b6..7212a8ae6b 100644
--- a/BaseTools/Plugin/LinuxGccToolChain/LinuxGccToolChain.py
+++ b/BaseTools/Plugin/LinuxGccToolChain/LinuxGccToolChain.py
@@ -22,45 +22,6 @@ class LinuxGccToolChain(IUefiBuildPlugin):
 
     def do_pre_build(self, thebuilder):
         self.Logger = logging.getLogger("LinuxGccToolChain")
-
-        #
-        # In order to keep this plugin backwards compatible with the older GCC5 toolchain tag,
-        # we support setting either the GCC5 or GCC set of env variables required
-        #
-        toolchain = "GCC"
-
-        #
-        # GCC - The non-x86 compilers need their paths set if available. To make this more stable, check for
-        # any substring of GCC in the TOOL_CHAIN_TAG to get the right compiler
-        #
-        if "GCC" in thebuilder.env.GetValue("TOOL_CHAIN_TAG"):
-            if "GCC5" in thebuilder.env.GetValue("TOOL_CHAIN_TAG"):
-                toolchain = "GCC5"
-
-            # Start with AARACH64 compiler
-            ret = self._check_aarch64(toolchain)
-            if ret != 0:
-                self.Logger.critical("Failed in check aarch64")
-                return ret
-
-            # Check arm compiler
-            ret = self._check_arm(toolchain)
-            if ret != 0:
-                self.Logger.critical("Failed in check arm")
-                return ret
-
-            # Check RISCV64 compiler
-            ret = self._check_riscv64(toolchain)
-            if ret != 0:
-                self.Logger.critical("Failed in check riscv64")
-                return ret
-
-            # Check LoongArch64 compiler
-            ret = self._check_loongarch64(toolchain)
-            if ret != 0:
-                self.Logger.critical("Failed in check loongarch64")
-                return ret
-
         return 0
 
     def _check_arm(self, toolchain):
