--- a/Silicon/NVIDIA/Library/DxeDtPlatformDtbLoaderLib/DxeDtPlatformDtbKernelLoaderLib.c
+++ b/Silicon/NVIDIA/Library/DxeDtPlatformDtbLoaderLib/DxeDtPlatformDtbKernelLoaderLib.c
@@ -761,7 +761,7 @@ DtPlatformLoadDtb (
   Status = gBS->CreateEventEx (
                   EVT_NOTIFY_SIGNAL,
                   TPL_CALLBACK,
-                  InstallFdt,
+                  UpdateFdt,
                   NULL,
                   &gEfiEventReadyToBootGuid,
                   &ReadyToBootEvent
